sbi/inference/trainers/npse/npse.py

```diff
@@ -363,20 +370,42 @@ def default_calibration_kernel(x):
                         batch[1].to(self._device),
                         batch[2].to(self._device),
                     )
+
+                    # For validation loss, we evaluate at a fixed set of times to reduce
+                    # the variance in the validation loss, for improved convergence
+                    # checks. We evaluate the entire validation batch at all times, so
+                    # we repeat the batches here to match.
+                    val_batch_size = theta_batch.shape[0]
+                    times_batch = validation_times.shape[0]
+                    theta_batch = theta_batch.repeat(
+                        times_batch, *([1] * (theta_batch.ndim - 1))
+                    )
+                    x_batch = x_batch.repeat(times_batch, *([1] * (x_batch.ndim - 1)))
+                    masks_batch = masks_batch.repeat(
+                        times_batch, *([1] * (masks_batch.ndim - 1))
+                    )
+
+                    validation_times_rep = validation_times.repeat_interleave(
+                        val_batch_size, dim=0
+                    )
+
                     # Take negative loss here to get validation log_prob.
                     val_losses = self._loss(
-                        theta_batch,
-                        x_batch,
-                        masks_batch,
-                        proposal,
-                        calibration_kernel,
+                        theta=theta_batch,
+                        x=x_batch,
+                        masks=masks_batch,
+                        proposal=proposal,
+                        calibration_kernel=calibration_kernel,
+                        times=validation_times_rep,
                         force_first_round_loss=force_first_round_loss,
                     )
+
+                    # print("val_losses: ", val_losses.shape)
                     val_loss_sum += val_losses.sum().item()
 
             # Take mean over all validation samples.
             val_loss = val_loss_sum / (
-                len(val_loader) * val_loader.batch_size  # type: ignore
+                len(val_loader) * val_loader.batch_size * times_batch  # type: ignore
             )
 
             # NOTE: Due to the inherently noisy nature we do instead log a exponential
```

Comment:
this comment is difficult to follow. can we give more details here? or move it to the docstring of train?